apps/x509.c

```diff
@@ -711,9 +711,12 @@ int x509_main(int argc, char **argv)
                        : "Certificate request self-signature did not match the contents\n");
             goto err;
         }
+      
         BIO_printf(out, "Certificate request self-signature ok\n");
-
-        print_name(out, "subject=", X509_REQ_get_subject_name(req));
+      
+        if (out != NULL){
+           print_name(out, "subject=", X509_REQ_get_subject_name(req));
+        }
```

Comment:
This looks like the wrong fix to me. `out` will never by non-NULL. Probably a better approach would be to move the `out = bio_open_default(outfile, 'w', outformat);` code from later on in this file, to before where `out` is used here. Another possibility is to just delete this lines since they never worked anyway.

Reply:
These lines used bio_err in 1.1.1 and should be changed to use it again in 3.0. It's simply a regression from 3.0.